### ISSUE TITLE
Support for the test.ai ML element locator

### ISSUE THREAD

**Issue by rakeshnambiar** [serenity-bdd/serenity-core#2676]
In order to make use of this utility we have to :

- An option to pass the API key to the driver which can read from the config file
  `driver = test_ai.TestAIDriver(driver, “API_KEY”)`

- An extra field for each element to pass a unique name along with the locator
  `driver.find_element_by_css_selector("input.countrycode", element_name='elem_country_code')`

**Comment by ricardorlg** [comment]
I could take this, as soon as the test.ai java library for this is released.

**Comment by rakeshnambiar** [comment]
@wakaleo @ricardorlg Please can we make some progress with this?

**Comment by wakaleo** [comment]
@rakeshnambiar there is no SLA for open source bug fixes, or even any guarantee that a particular ticket will be addressed (remember, you are asking folk to volunteer their personal and family time to do work for you). If this feature is important to you, and you would like it to progress faster, you have a few options:
  1) You can do the work yourself (or pay someone to do it for you) and raise a pull request;
  2) You can get your company to purchase a commercial support package (there are several options available that I can provide you with offline if you would like to explore this option)
  3) Alternatively, since this is integration with a commercial product, you could ask the vendor (test.ai) to sponsor this work.

**Comment by ricardorlg** [comment]
I'm working on this in my free time, I don't have an estimated time to finish, It will be ready when it's ready.

**Comment by rakeshnambiar** [comment]
Hi @wakaleo @ricardorlg thanks for letting me the situation. I have also asked the `test.ai` in case they are willing with option#3. Considering @ricardorlg already working on it in parallel, better I will wait.

**Comment by rakeshnambiar** [comment]
@ricardorlg I knew you are busy with many things and appreciate your efforts. Just wanted to follow-up with the latest status of this request.

**Comment by wakaleo** [comment]
If test.ai would like to implement an integration, or fund some work (through either commercial support package or sponsoring for example), that would probably be the fastest way to get this feature implemented.

**Comment by rakeshnambiar** [comment]
@cnavrides @alexander-wu-aw Please could you able to help with it?

**Comment by cnavrides** [comment]
@rakeshnambiar yes we can add an item to our board to look into this integration

**Comment by rakeshnambiar** [comment]
@cnavrides Thanks a lot for your support. I created the issue here https://github.com/dev-tools-ai/java-selenium-sdk/issues/1 Let me know in case there's a separate board available to log such requests.
